### PR TITLE
OS#13641974 Asm.js internal call thunk

### PR DESCRIPTION
**Pull request by Cellule**
Refactor CallAsmJsFunction to offset most of the work in C++ instead of the assembly thunks

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/3714)

### CODE REVIEW

**Comment by Cellule** [comment]
Ping

File: lib/Runtime/Library/amd64/JavascriptFunctionA.asm [hunk 1]
```diff
@@ -219,198 +219,118 @@ ifdef _ENABLE_DYNAMIC_THUNKS
...
         .pushreg rsi
         push rdi
         .pushreg rdi
-        push r12
-        .pushreg r12
-        push r13
-        .pushreg r13
+        ; push r12 unused
+        ; .pushreg r12
+        ; push r13 unused
```
Comment (MikeHolman) [inline lib/Runtime/Library/amd64/JavascriptFunctionA.asm]: nit: remove these comments

Comment (Cellule) [reply lib/Runtime/Library/amd64/JavascriptFunctionA.asm]: Wouldn't it be better to leave it so if someone starts using r12,r13,... they can see that they're not currently saved by the callee ?

Comment (MikeHolman) [reply lib/Runtime/Library/amd64/JavascriptFunctionA.asm]: i guess so, i don't really care much either way so i'm fine leaving it if you think there is value.

---
In reply to: [139541099](https://github.com/Microsoft/ChakraCore/pull/3714#discussion_r139541099) [](ancestors = 139541099)

**Pull request merged**